### PR TITLE
Appoint-doctor-backend: First version of the API endpoints

### PR DESCRIPTION
**Pull request by AbdusSattar-70**
## Hello Reviewer,🎉

Taking a look at this project, five of us, @danutibe07, @lengushuru, @Tracy-miranja, @TaiwoEnoch, and @AbdusSattar-70  (me) worked hard on this project.
 ### [Project requirements link here](https://github.com/microverseinc/curriculum-final-capstone/blob/main/projects/business_requirements.md)
- [ ] We select `appointment with a doctor` as a project theme.

- [ ] We implemented the back-end for the CRUD operation with its API.
- [ ] We implemented the `devise-jwt` for proper safety and Created auth token.

- [ ] We created the front end with React meant to consume the API.

- [ ] The user logs in to the website, only by typing the username (a properly authenticated login is required if your group is made of 5 people).
- [ ] We implemented the proper Authentication for the application as we are five.

- [ ] On the main page, the user can see a list of doctors

- [ ] When the user selects a specific item, they can see the doctor's details

- [ ] When the user clicks the `add doctor)/Delete doctor` link in the navigation panel they can see a form for adding a new doctor or delete doctor page Only the admin can add/delete items(doctors)

- [ ] In the details page, the user can click the `appointment` button to make an appointment with a doctor 
- [ ] In the `navbar` we made the appointment button as well.

- [ ] Made the app responsive, creating both mobile and desktop versions.

- [Link to the Frontend](https://github.com/Tracy-miranja/appoint-doctor-frontend)
- [ API Documentation Here](https://documenter.getpostman.com/view/26957552/2s9XxsVGHT)
- [Kanban board link:](https://github.com/users/Tracy-miranja/projects/8)
- [Screenshot of the initial Stage](https://user-images.githubusercontent.com/114406879/256285403-e8a11387-6a54-4695-b302-6d08faaa85ea.png)
NOTE
We have deployed it online and it's production ready, if you want to run it on the local machine, comment out Line no. 33 and Line no 40 on Config -->` Puma.rb.` Thanks.😍

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
+<div align="center">
```
Comment (Meltrust) [inline README.md:1]: - [ ] Your project needs a `secret-key-base` created locally:

image

Kindly, add the instructions to create it to your readme:

```
1. Remove config/master.key and config/credentials.yml.enc if they exist.
2. Run in the terminal: EDITOR=code rails credentials:edit 
3. Close the editor that opens.
4. This command will create a new master.key and credentials.yml.enc if they do not exist.
```

That way, the database creation will work for all your users. 👍

Comment (Meltrust) [inline README.md:1]: - [ ] Kindly, include in your instructions the required db commands that your users will need to run your backend app: `rails db:create, rails db:migrate`. 👍

File: README.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+### Usage
+
+- Run the app with `rails s` in the terminal from the root folder of the project.
+
+### Test
+
+```sh
+  rspec spec/filename.rb
+```
```
Comment (Meltrust) [inline README.md:134]: - [ ]  You are using Rspec, but the "test" folder still exists, which is used by another test framework. Kindly consider deleting that folder if you are not using it. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @AbdusSattar-70,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  Nice code organization ✔️ 
-  API is working well✔️ 
-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: README.md (hunk 2 above)
Comment (AbdusSattar-70) [reply README.md:134]: Dear @Meltrust Thanks for your valuable suggestion, We will add everything that you suggested.🎉😍

**Review by Meltrust — APPROVED** [review]
_note: This review outcome only applies to the backend project._

#### Hi @AbdusSattar-70,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Readme has been improved✔️ 
- Superfluous test folder has been removed✔️ 
- Nice job✔️ 



Cheers and Happy coding!👏👏👏


------

**Comment by AbdusSattar-70** [comment]
Hello @Meltrust Thanks for the approval 🎉😍

**Pull request merged**